Tests/ConfigurationTests/ConfigurationManagerTests.swift

```diff
@@ -0,0 +1,207 @@
+        await managerA.refreshNow()
+        await fulfillment(of: [e!], timeout: 2)
+
+        XCTAssertEqual(managerB.dependencyProvider.privacyConfigData, configData)
+        XCTAssertEqual(managerB.dependencyProvider.privacyConfigEtag, HTTPURLResponse.testEtag)
+    }
+
+    func testWhenManagerBReceivesNewDataManagerAHasDataAfter304Response() async throws {
+        let managerA = makeConfigurationManager()
+        let managerB = makeConfigurationManager()
+
+        var e: XCTestExpectation? = expectation(description: "ConfigManager B updated")
+        managerB.onDependenciesUpdated = {
+            e?.fulfill()
+            e = nil
+        }
+
+        var configData = Data("Privacy Config".utf8)
+        MockURLProtocol.requestHandler = { _ in (HTTPURLResponse.ok, configData) }
+        await managerA.refreshNow()
+        await fulfillment(of: [e!], timeout: 2)
+
+        XCTAssertEqual(managerB.dependencyProvider.privacyConfigData, configData)
+        XCTAssertEqual(managerB.dependencyProvider.privacyConfigEtag, HTTPURLResponse.testEtag)
+
+        e = expectation(description: "ConfigManager A updated")
+        managerB.onDependenciesUpdated = nil
+        managerA.onDependenciesUpdated = {
+            e?.fulfill()
+            e = nil
+        }
+
+        configData = Data("Privacy Config 2".utf8)
+        MockURLProtocol.requestHandler = { _ in (HTTPURLResponse.ok, configData) }
+        await managerB.refreshNow()
+        await fulfillment(of: [e!], timeout: 2)
+
+        XCTAssertEqual(managerA.dependencyProvider.privacyConfigData, configData)
+        XCTAssertEqual(managerA.dependencyProvider.privacyConfigEtag, HTTPURLResponse.testEtag)
+
+        MockURLProtocol.requestHandler = { _ in (HTTPURLResponse.notModified, nil) }
+        await managerA.refreshNow()
+        XCTAssertEqual(managerA.dependencyProvider.privacyConfigData, configData)
+        XCTAssertEqual(managerA.dependencyProvider.privacyConfigEtag, HTTPURLResponse.testEtag)
+    }
+
+    func testWhenManagerBReceivesAnErrorItKeepsDataFromManagerA() async throws {
+        let managerA = makeConfigurationManager()
+        let managerB = makeConfigurationManager()
+
+        var e: XCTestExpectation? = expectation(description: "ConfigManager B updated")
+        managerB.onDependenciesUpdated = {
+            e?.fulfill()
+            e = nil
+        }
+
+        let configData = Data("Privacy Config".utf8)
+        MockURLProtocol.requestHandler = { _ in (HTTPURLResponse.ok, configData) }
+        await managerA.refreshNow()
+        await fulfillment(of: [e!], timeout: 2)
+
+        XCTAssertEqual(managerB.dependencyProvider.privacyConfigData, configData)
+        XCTAssertEqual(managerB.dependencyProvider.privacyConfigEtag, HTTPURLResponse.testEtag)
+
+        MockURLProtocol.requestHandler = { _ in (HTTPURLResponse.internalServerError, nil) }
+        await managerB.refreshNow()
```

Comment:
Just to be safe we could also assert against the `lastRequest` property of `MockURLProtocol`, to prove for sure that the request is being made and thus the behavior is correct.